resources/testdriver.js

```diff
@@ -137,27 +137,64 @@
         }
     };
 
-    window.test_driver_internal = {
+    var manual = {
         /**
-         * Triggers a user-initiated click
+         * This flag should be set to `true` by any code which implements the
+         * internal methods defined below for automation purposes. Doing so
+         * allows the library to signal failure immediately when an automated
+         * implementation of one of the methods is not available.
+         */
+        in_automation: false,
```

Comment:
Defaulting to manual means we'll have to react quickly after this is imported into Chromium to see what broke. @LukeZielinski can you try running this in Chromium CI and prepare any adaptation we need?

Reply:
I wonder if we can set the default to true initially so that when this gets imported, everything works the same way. That can give us time to override it to true in the Chromium testdriver. A follow-up PR can then flip the default to false. Does that sound reasonable?

Reply:
I don't think there is any risk to Chrome in landing this with `in_automation` set to `false` by default.

The `in_automation` flag controls how the manual methods behave; its purpose is to account for future extensions. The value will be relevant when a new method is added to `testdriver.js` that the consumer has *not* implemented. In those cases, the "manual" implementation will detect that it is running in automation and return a rejected Promise as "not implemented".

Failing to set this flag will only result in test timeouts instead of test failures (and again, only when new methods are available in `testdriver.js`).

I don't know where/how Chrome currently implements `test_driver_internal`, but I believe it only lacks the `action_sequence` method. We have no plans to implement a manual version of that method (since communicating the expected input would be very difficult), so that will always report "not implemented", regardless of the value of `in_automation`.

Reply:
Chrome implementation of `test_driver_internal` can be found here: https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/resources/testdriver-vendor.js?l=78&rcl=57d76f99672823e61bb01a8d132c5c663caf899b

Reply:
Yep, you're right, so nevermind - Chrome implements all of the apis so we don't fall back to the timeout case.

Reply:
@LukeZielinski prepared for this in https://chromium-review.googlesource.com/c/chromium/src/+/1352571 anyway, thanks Luke!